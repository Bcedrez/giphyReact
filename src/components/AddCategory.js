import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCatergory = ({setCategorias}) => {


    const [valorInput, SetValorInput] = useState('')

    const handleValorInput = (e) => {
        SetValorInput(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(valorInput.trim().length > 1){
            setCategorias(categorias =>[valorInput,...categorias])
            SetValorInput('')
        }
        
        
    }

    return (

      
        <form onSubmit={handleSubmit}>
            <input
            placeholder="Buscar Gif"
            type="text"
            value={valorInput}
            onChange={handleValorInput}
            />
        </form>
    );
}

AddCatergory.propTypes = {
    setCategorias : PropTypes.func.isRequired
}