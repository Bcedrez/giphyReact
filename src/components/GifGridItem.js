import React from "react";
import PropTypes from 'prop-types';


const GifGridItem = ({id,titulo,url})=>{
    // console.log({id,titulo,url})
    return (
        <div className="card animate__animated animate__zoomIn">
           <img src={url} alt={titulo} />
           <p>{titulo}</p>
        </div>
    );
}

GifGridItem.propTypes = {
    titulo: PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
}


export default GifGridItem