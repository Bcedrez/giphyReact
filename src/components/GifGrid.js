import React  from "react";

import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks/useFechGifs";
import  GifGridItem  from "./GifGridItem";



export const GifGrid = ({categoria})=>{
    
    const {data:imagenes,loading} = useFetchGifs(categoria);   

    return (
        <>
        <h3 className="animate__animated animate__rotateInDownLeft">{categoria}</h3>

        {loading && <p>loading...</p>}
        
        <div className="card-grid">  
            
                {
                    imagenes.map(imagen =>{
                        return <GifGridItem 
                                    key={imagen.id} 
                                    {...imagen}
                                   
                                     />
                    })
                }     
            
        </div>
       </> 
    )
}

GifGrid.propTypes = {
    categoria : PropTypes.string.isRequired
}