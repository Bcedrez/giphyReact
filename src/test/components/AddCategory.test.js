import React from 'react';

import { shallow } from "enzyme"
import { AddCatergory } from "../../components/AddCategory"


describe('Pruebas en <AddCategory />',()=>{

    const setCategories = jest.fn();
    let wrapper
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCatergory setCategorias={setCategories}/>);
    })

    test('Debe renderizarse correctamente',()=>{
        
        expect(wrapper).toMatchSnapshot()
    });

    test('Debe de cambiar la caja de texto',()=>{

        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        input.simulate('change',{target : {value}});
    });

    test('No debe ejecutar el submit',()=>{

        wrapper.find('form').simulate('submit',{preventDefault(){}});
        
        expect(setCategories).not.toHaveBeenCalled();

    });

    test('Debe de llamar SetCategories y limpiear caja de texto',()=>{

        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        input.simulate('change',{target : {value}});

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        expect(setCategories).toHaveBeenCalledWith( expect.any(Function));

        expect(input.props().value).toBe('');


    });

})