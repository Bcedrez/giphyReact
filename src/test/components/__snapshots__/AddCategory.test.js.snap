// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe renderizarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar Gif"
    type="text"
    value=""
  />
</form>
`;
